src/engine/runtime.js

```diff
@@ -51,30 +51,58 @@ const defaultExtensionColors = ['#0FBD8C', '#0DA57A', '#0B8E69'];
 const ArgumentTypeMap = (() => {
     const map = {};
     map[ArgumentType.ANGLE] = {
-        shadowType: 'math_angle',
-        fieldType: 'NUM'
+        shadow: {
+            type: 'math_angle',
+            // We specify fieldNames here so that we can pick
+            // create and populate a field with the defaultValue
+            // specified in the extension.
```

Comment:
I'm not sure I understand this comment... specifically the "so that we can pick create and populate" part. Is there a typo there or am I just failing to parse it right?

Reply:
Ah, I was differentiating "creating a field" e.g. laying out space on the block for the correct field, vs. "populating a field" as in filling the empty text input (e.g. for number fields) with the number `10` for `move 10 steps`